framework/Assembler.py

```diff
@@ -122,58 +128,40 @@ def _readMoreXML(self, xmlNode):
+      for token in self._requiredAsmbObject[1]:
         testObjects[token] = 0
       found = dict.fromkeys(testObjects.keys(),False)
       found, testObjects = self._readAssemblerObjects(xmlNode, found, testObjects)
       for subNode in xmlNode:
         found, testObjects = self._readAssemblerObjects(subNode, found, testObjects)
-      for token in self.requiredAssObject[1][0]:
-        if not found[token] and not str(self.requiredAssObject[1][1][self.requiredAssObject[1][0].index(token)]).strip().startswith('-'):
-          self.raiseAnError(IOError,'the required object ' +token+ ' is missed in the definition of the '+self.type+' Object! Required objects number are :'+str(self.requiredAssObject[1][1][self.requiredAssObject[1][0].index(token)]))
-      # test the objects found
-      else:
-        for cnt,toObjectName in enumerate(self.requiredAssObject[1][0]):
-          numerosity = str(self.requiredAssObject[1][1][cnt])
-          if numerosity.strip().startswith('-'):
-          # optional
-            if toObjectName in testObjects.keys():
-              if testObjects[toObjectName] is not 0:
-                numerosity = numerosity.replace('-', '').replace('n',str(testObjects[toObjectName]))
-                if testObjects[toObjectName] != int(numerosity):
-                  self.raiseAnError(IOError,'Only '+numerosity+' '+toObjectName+' object/s is/are optionally required. Block '+self.name + ' got '+str(testObjects[toObjectName]) + '!')
-          else:
-            # required
-            if toObjectName not in testObjects.keys():
-              self.raiseAnError(IOError,'Required object/s "'+toObjectName+'" not found. Block '+self.name + '!')
-            else:
-              numerosity = numerosity.replace('n',str(testObjects[toObjectName]))
-              if testObjects[toObjectName] != int(numerosity):
-                self.raiseAnError(IOError,'Exactly {n} <{t}> nodes are required for <{c}> "{m}". Got {g}!'
-                                 .format(n=numerosity,
-                                         t=toObjectName,
-                                         c=self.type,
-                                         m=self.name,
-                                         g=testObjects[toObjectName]))
-    if '_localReadMoreXML' in dir(self):
+      for i,token in enumerate(self._requiredAsmbObject[1]):
+        quantity = self._requiredAsmbObject[2][i]
+        if not InputData.checkQuantity(quantity, testObjects[token]):
+          self.raiseAnError(IOError, 'the object '+token+' has wrong quantity Expected: '+str(quantity)+' Found: '+str(testObjects[token])+ ' in block '+self.name)
+
+    if '_handleInput' in dir(self) and self._handleInput.__func__.__qualname__.split(".")[0] == self.__class__.__name__:
+      #_handleInput in class and not from superclass
+      print(self, self.getInputSpecification, self.getInputSpecification.__func__.__qualname__, self._handleInput, self._handleInput.__func__.__qualname__)
```

Comment:
Is this print statement meant to be retained?

Reply:
Commented out (useful for debugging but does not need to be printed most times)